src/polynomial.rs

```diff
@@ -59,6 +60,69 @@ impl Polynomial {
     pub fn to_vec(&self) -> Vec<Fr> {
         self.elements.clone()
     }
+    pub fn ifft_on_elements(&mut self) -> Result<(), PolynomialError> {
+        let ifft_result = Self::ifft(&self.to_vec());
+        match ifft_result {
+            Ok(ifft_result) => {
+                self.elements = ifft_result;
+                Ok(())
+            },
+            Err(e) => Err(e),
+        }
+    }
+
+    pub fn fft_on_elements(&mut self) -> Result<(), PolynomialError> {
```

Comment:
the above 2 functions can be combined.

Reply:
combined in df504078e06c54f48c23b1006916455fe75fd0a0